stan/math/rev/core/var.hpp

```diff
@@ -371,10 +371,23 @@ class var_value<T, internal::require_matrix_var_value<T>> {
    * @param other the value to assign
    * @return this
    */
-  template <typename S, require_assignable_t<value_type, S>* = nullptr,
+  template <typename S,
```

Comment:
What's the purpose of this change? I don't recall seeing a note in the PR.

Reply:
Oh I need to write docs for this. This change handles `vari_type`s  where the inner type of Eigen matrices are assignable to one another, but the actual `vari_type` is not. So things like `vari_value<Matrix<double, -1, 1>>` and `vari_value<Matrix<double, 1, -1>>`. As pointers those are not assignable to one another, but their inner matrix types are. So the `var` has to make a new `vari` and assign the inner matrices to that new `vari`.